.github/workflows/functional-baremetal.yaml

```diff
@@ -91,9 +96,7 @@ jobs:
           OS_BRANCH: ${{ matrix.openstack_version }}
       - name: Generate logs on failure
         run: ./script/collectlogs
-        if: failure()
```

Comment:
we did this to save log artifacts, we found that it wasn't super useful to store all these files if the jobs passed. WDYT?

Reply:
yeah, I just removed in case things failed again I wanted to double check the pip-freeze from a green job (sometimes it makes sense to collect a few things I would say)

Reply:
cool, yeah let's just remove it.